### PR TITLE
implement functional set of integers. add tests for all functions.

### PR DESCRIPTION
**Pull request by LizaDashevski**

### CODE REVIEW

File: ex2/FunSetSuite.scala [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+import org.scalatest.junit.JUnitRunner
+
+/**
+ * This class is a test suite for the methods in object FunSets. To run
+ * the test suite, you can either:
+ *  - run the "test" command in the SBT console
+ *  - right-click the file in eclipse and chose "Run As" - "JUnit Test"
+ */
+@RunWith(classOf[JUnitRunner])
+class FunSetSuite extends FunSuite {
```
Comment (oridanus) [inline ex2/FunSetSuite.scala:16]: usually, GitHub shows the test after the "real" code, I wonder how it does it...
maybe you can you try to rename it to something like FunSetTest? or have it under test package? and then see if the tests come last...

**Review by oridanus — COMMENTED** [review]
will try to go over the logic again...
all tests pass?

File: ex2/FunSetSuite.scala (hunk 1 above)
Comment (LizaDashevski) [reply ex2/FunSetSuite.scala:16]: Ok, so I didn't want to use the course actual files and folder structure, due to this code not being mine, so I just pushed the code I wrote.

All tests pass :)

File: ex2/FunSets.scala [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    }
+    //iter from initial val
+    iter(-bound)
+  }
+
+  /**
+    * Returns whether there exists a bounded integer within `s`
+    * that satisfies `p`.
+    */
+  def exists(s: Set, p: Int => Boolean): Boolean = {!forall(s, elem => !p(elem))}
```
Comment (oridanus) [inline ex2/FunSets.scala:74]: try not to use double negations.
maybe clearer will be to extract the `iter` method from `forall` and use it here

File: ex2/FunSetSuite.scala (hunk 1 above)
Comment (oridanus) [reply ex2/FunSetSuite.scala:16]: can you try to rename to `FunSetTest` to see if github put it second to the "real" code?

**Pull request closed without merging**